src/ast/async_event_types.h

```diff
@@ -59,5 +59,19 @@ struct Time
   }
 } __attribute__((packed));
 
+struct Buf
+{
+  uint8_t length;
+  uint8_t* content;
```

Comment:
It's a bit misleading to make `content` a `uint8_t*` here because what you actually mean here is `uint8_t content[VARIABLE_VALUE]`.

Typically how this is done in the kernel is with a flexible length array. I think we can use that here too.

Maybe like this:
```
struct Buf
{
  uint8_t length;
  uint8_t content[];
} __attribute__((packed));
```

Then you alloca a `fixed_buffer_length + sizeof(uint8_t)` length buffer.

This way we can properly cache the llvm types. And also be less misleading.